mmcv/runner/hooks/logger/neptune.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) Open-MMLab. All rights reserved.
+from ...dist_utils import master_only
+from ..hook import HOOKS
+from .base import LoggerHook
+
+
+@HOOKS.register_module()
+class NeptuneLoggerHook(LoggerHook):
+
+    def __init__(self,
+                 init_kwargs=None,
+                 interval=10,
+                 ignore_last=True,
+                 reset_flag=True,
+                 with_step=True,
+                 by_epoch=True):
+        """Class to log metrics to NeptuneAI.
+
+        It requires `neptune-client` to be installed.
+
+        Args:
+            init_kwargs:
+                project (str): Name of a project in a form of
+                    namespace/project_name. If None, the value of
+                    NEPTUNE_PROJECT environment variable will be taken.
+                api_token (str): User’s API token.
+                    If None, the value of NEPTUNE_API_TOKEN environment
+                    variable will be taken. Note: It is strongly recommended
+                    to use NEPTUNE_API_TOKEN environment variable rather than
+                    placing your API token in plain text in your source code.
+                name (str, optional, default is 'Untitled'): Editable name of
+                    the run. Name is displayed in the run's Details and in
+                    Runs table as a column.
+                Check https://docs.neptune.ai/api-reference/neptune#init for
+                    more init arguments.
+            interval (int): Logging interval (every k iterations).
+            ignore_last (bool): Ignore the log of last iterations in each epoch
+                if less than `interval`.
+            reset_flag (bool): Whether to clear the output buffer after logging
+            by_epoch (bool): Whether EpochBasedRunner is used.
+
+        .. _NeptuneAI:
+            https://docs.neptune.ai/you-should-know/logging-metadata
+        """
+        super(NeptuneLoggerHook, self).__init__(interval, ignore_last,
+                                                reset_flag, by_epoch)
+        self.import_neptune()
+        self.init_kwargs = init_kwargs
+        self.with_step = with_step
+
+    def import_neptune(self):
+        try:
+            import neptune.new as neptune
+        except ImportError:
+            raise ImportError(
+                'Please run "pip install neptune-client" to install neptune')
+        self.neptune = neptune
+        self.run = None
+
+    @master_only
+    def before_run(self, runner):
+        if self.neptune is None:
```

Comment:
why we check the value of `self.neptune` again?

Reply:
i took wandb hooker as reference:
https://github.com/open-mmlab/mmcv/blob/b8c09f3b8449f2771e92a6a671027359d2c2707e/mmcv/runner/hooks/logger/wandb.py#L35

Reply:
This line may be redundant. You could try again after removing the line.

Reply:
@zhouzaida done.